### PR TITLE
Convert service model to typescript

### PR DESCRIPTION
**Pull request by sradevski**
Change-type: patch
Signed-off-by: Stevche Radevski <stevche@balena.io>

---
##### Contributor checklist

- [ ] Includes tests
- [ ] Includes typings
- [ ] Includes updated documentation
- [ ] Includes updated build output

### CODE REVIEW

File: lib/models/service.ts [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+			getResourceId: (id: number) => get(id, { $select: 'id' }).get('id'),
+			ResourceNotFoundError: errors.BalenaServiceNotFound,
+		},
+	);
+
+	exports = {};
+
+	// Not exported for now, but we could document & export in future
+	// if there are external use cases for this.
+	const get = (id: number, options: PineOptionsFor<Service>) => {
```
Comment (thgreasi) [inline lib/models/service.ts]: How about:
```suggestion
	const get = (id: number, options: PineOptionsFor<Service> = {}) => {
```
Applies in the rest of the file as well.

File: lib/models/service.ts [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+	 * @memberof balena.models.service
+	 *
+	 * @param {String|Number} nameOrSlugOrId - application name (string), slug (string) or id (number)
+	 * @param {Object} [options={}] - extra pine options to use
+	 * @fulfil {Object[]} - services
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.service.getAllByApplication('MyApp').then(function(services) {
+	 * console.log(services);
```
Comment (thgreasi) [inline lib/models/service.ts]: It seems that the indentation broke here, making the README examples to have zero indentation.
Let's re-indent them plz.
Applies to the two other examples right below.

File: lib/models/service.ts [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+	 * balena.models.service.getAllByApplication('MyApp', function(error, services) {
+	 * if (error) throw error;
+	 * console.log(services);
+	 * });
+	 */
+	exports.getAllByApplication = function(
+		nameOrSlugOrId: string | number,
+		options: PineOptionsFor<Service>,
+		callback?: (error?: Error, result?: Service) => void,
+	): Promise<Service> {
```
Comment (thgreasi) [inline lib/models/service.ts]:
```suggestion
	): Promise<Service[]> {
```

File: lib/models/service.ts [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+	 * @example
+	 * balena.models.service.getAllByApplication('MyApp', function(error, services) {
+	 * if (error) throw error;
+	 * console.log(services);
+	 * });
+	 */
+	exports.getAllByApplication = function(
+		nameOrSlugOrId: string | number,
+		options: PineOptionsFor<Service>,
+		callback?: (error?: Error, result?: Service) => void,
```
Comment (thgreasi) [inline lib/models/service.ts]:
```suggestion
		callback?: (error?: Error, result?: Service[]) => void,
```

File: lib/models/service.ts [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+		 * @example
+		 * balena.models.service.var.getAllByApplication('MyApp', function(error, vars) {
+		 * 	if (error) throw error;
+		 * 	console.log(vars)
+		 * });
+		 */
+		getAllByApplication(
+			nameOrSlugOrId: string | number,
+			options: PineOptionsFor<ServiceEnvironmentVariable>,
+			callback?: (error?: Error, result?: ServiceEnvironmentVariable) => void,
```
Comment (thgreasi) [inline lib/models/service.ts]:
```suggestion
			callback?: (error?: Error, result?: ServiceEnvironmentVariable[]) => void,
```

**Comment by thgreasi** [comment]
Also let's rename the commit to something like
> Convert the service model to typescript

since the CHANGELOG is the way that we communicate changes to our users :)

File: lib/models/service.ts (hunk 1 above)
Comment (sradevski) [reply lib/models/service.ts]: I just used decaffeinate and didn't really think of doing this

File: lib/models/service.ts (hunk 2 above)
Comment (sradevski) [reply lib/models/service.ts]: the linter was complaining about the original formatting, but I managed to get it fixed now.

**Pull request merged**